### DISCUSSION TITLE
Pseudo replay id for battle replays

### DISCUSSION THREAD

**Discussion by ville-j** [elmadev/elmaonline-site#512] [Feature discussion]
Because battle winning replays are stored differently from the uploaded replays it creates some problems linking to them etc. Could we perhaps use a "pseudo id" for battle winning replays that are prefixed with "b-" or some other string? For example [https://elma.online/r/b-169159](https://elma.online/r/b-169159) would display the normal replay page with the battle winning replay that can be viewed in the battle page [https://elma.online/battles/169159](https://elma.online/battles/169159). This way it would be possible to comment and rate battle replays too, and it would be more consistent experience to link to these for example from search results etc. I think handling all this could be done purely on the frontend side (unless the db requires a row with the replay id to exist) but I might be mistaken. Has this kind of thing ever been discussed?

**Comment by sunehs** [comment]
Some kind of merging of uploaded and battle recs has been talked about a number of times, as recent as yesterday in discord.

I started on testing out some changes to the api for getting replays for a levelindex to include battle replays. So lists like the replays on a level page is more complete. That would also require something like what you're suggesting on the replay page for those links to work. I even used the same b-{BattleIndex} string :)

Replay comments and ratings are bound to an int ReplayIndex, so would need some change to that, but that should be doable. Maybe a secondary field or just making the ReplayIndex to a longer string. I don't think that should cause any issues.
